src/Microsoft.AspNetCore.Routing/Patterns/RoutePatternFactory.cs

```diff
@@ -451,12 +449,7 @@ public static RoutePatternPathSegment Segment(params RoutePatternPart[] parts)
                 throw new ArgumentNullException(nameof(parts));
             }
 
-            return SegmentCore(parts);
-        }
-
-        private static RoutePatternPathSegment SegmentCore(IEnumerable<RoutePatternPart> parts)
-        {
-            return new RoutePatternPathSegment(parts.ToArray());
```

Comment:
Can you put this back and have it accept an array? I know it's boilerplate but it's symetry with the other things that are here.

Reply:
Done.